File: example-base-28/nodejs-utilities-playground-v2/src/dsa/strings/valid-parentheses.js
function isValidParentheses(value) {
  const stack = [];
  const pairs = new Map([[")", "("], ["]", "["], ["}", "{"]]);
  for (const char of value) {
    if (["(", "[", "{"].includes(char)) stack.push(char);
    else if (pairs.has(char) && stack.pop() !== pairs.get(char)) return false;
  }
  return stack.length === 0;
}
module.exports = { isValidParentheses };

File: example-base-29/python-algorithms-playground-v2/algorithms/strings/valid_parentheses.py
def is_valid_parentheses(s):
    pairs={')':'(',']':'[','}':'{'}; stack=[]
    for ch in s:
        if ch in '([{': stack.append(ch)
        elif ch in pairs:
            if not stack or stack.pop()!=pairs[ch]: return False
    return not stack

if __name__ == "__main__": print(is_valid_parentheses('({[]})'))
